etor/src/main/java/gov/hhs/cdc/trustedintermediary/etor/ruleengine/RuleEngine.java

```diff
@@ -0,0 +1,60 @@
+package gov.hhs.cdc.trustedintermediary.etor.ruleengine;
+
+import gov.hhs.cdc.trustedintermediary.wrappers.Logger;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+import javax.inject.Inject;
+import org.hl7.fhir.instance.model.api.IBaseResource;
+
+/** Manages the application of rules loaded from a definitions file using the RuleLoader. */
+public class RuleEngine {
+
+    private final Path RULES_DEFINITIONS_PATH =
+            Path.of("../etor/src/main/resources/rule_definitions.json");
+    private boolean rulesLoaded = false;
+
+    private static final RuleEngine INSTANCE = new RuleEngine();
+
+    final List<Rule> rules = new ArrayList<>();
+
+    @Inject Logger logger;
+    @Inject RuleLoader ruleLoader;
+
+    private RuleEngine() {}
+
+    public static RuleEngine getInstance() {
+        return INSTANCE;
+    }
+
+    public void unloadRules() {
+        rulesLoaded = false;
+        rules.clear();
+    }
+
+    public void ensureRulesLoaded() {
+        if (!rulesLoaded) {
```

Comment:
Should this be a boolean or just a check if there's any rules in the list

Reply:
Changed it, it's now a check with one less var to keep track of.